packages/flutter_availability/lib/src/ui/view_models/week_template_view_models.dart

```diff
@@ -0,0 +1,80 @@
+import "package:flutter_availability/src/ui/view_models/template_daydata_view_model.dart";
+import "package:flutter_availability_data_interface/flutter_availability_data_interface.dart";
+
+/// View model to represent the data during the modification of a week template
+class WeekTemplateViewModel {
+  /// Constructor
+  const WeekTemplateViewModel({
+    this.data = const {},
+    this.id,
+    this.name,
+    this.color,
+  });
+
+  /// Create a [WeekTemplateViewModel] from a [AvailabilityTemplateModel]
+  factory WeekTemplateViewModel.fromTemplate(
+    AvailabilityTemplateModel template,
+  ) {
+    var data = template.templateData as WeekTemplateData;
+    return WeekTemplateViewModel(
+      id: template.id,
+      name: template.name,
+      color: template.color,
+      data: {
+        for (var day in WeekDay.values)
+          day: data.data.containsKey(day)
+              ? DayTemplateDataViewModel.fromDayTemplateData(
+                  data.data[day]!,
+                )
+              : const DayTemplateDataViewModel(),
+      },
+    );
+  }
+
+  /// The identifier for this template
+  final String? id;
+
+  /// The name by which the template can be visually identified
+  final String? name;
+
+  /// The color by which the template can be visually identified
+  final int? color;
+
+  /// The data for each day of the week
+  final Map<WeekDay, DayTemplateDataViewModel> data;
+
+  /// Whether the data is valid for saving
+  bool isValid() => data.values.every((e) => e.isValid);
```

Comment:
A weektemplate is valid if:
- At least 1 day is valid.
- No days are invalid

A day is valid:
- If the start is not null and the end is not null
- The start is before the end.
- All provided breaks are valid.

A break is valid:
- If the start is before the end
- If the duration is equal or lower to the difference between start and end